README.md

```diff
@@ -28,6 +32,121 @@ Project is written in TypeScript and provides type safety out of the box. No Flo
+## Overview
+
+This library has two goals:
+
+1. Make it possible to introduce reactivity to React function components
+2. Make it possible to have local observable state in React function components
+
+This library does _not_ support class based components. If you need class based component support, use [mobx-react](https://github.com/mobxjs/mobx-react), which wraps this library.
+
+### Adding reactivity to React components
+
+There are three ways to add reactivity to components: `observer(fn)`, `<Observer>{renderFn}</Observer>` and `useObserver(renderFn)`.
+In most cases, there difference doesn't matter that much, but here is an overview of the three different approaches:
+
+```javascript
+import { observable } from "mobx"
+import { Observer, useObserver, observer }  from "mobx-react-lite"
+
+const person = observable({
+    name: "John"
+})
+
+const P1 = observer(function P1({ person }) {
+    return <h1>{person.name}</h1>
+})
+
+const P2 = ({ person }) => (
+    <Observer>{() => <h1>{person.name}</h1>}</Observer>
+)
+
+const P3 = ({ person }) => {
+    return useObserver(() => (
+        <h1>{person.name}</h1>
+    ))
+})
+
+ReactDOM.render(<div>
+    <P1 person={person} />
+    <P2 person={person} />
+    <P3 person={person} />
+</div>)
+
+setTimeout(() => {
+    person.name = "Jane"
+}, 1000)
+```
+
+When using `observer`, the component `props` will be made automatically observable (so that they can be used in reactions and computed values), and `React.memo` will be applied automatically to the comment.
+
+`useObserver` keeps the component tree flat as it doesn't introduce a higher order component, but to make props observable, the [`useAsObservableSource`](#useasobservablesourcetstate-t-t) hook should be used.
+
+`<Observer>` is slightly better optimized compared to `useObserver`, but does introduce an additional component in the component tree.
+
+### Introducing local observable state
+
+Local observable state can be introduced by using the [`useLocalStore`](#uselocalstoretinitializer---t-t) hook, that runs once to create an observable store. A quick example would be:
+
+```javascript
+import { useLocalStore, useObserver } from "mobx-react-lite"
+
+const P3 = () => {
+    const todo = useLocalStore(() => ({
+        title: "Test",
+        done: true,
+        toggle() {
+            this.done = !this.done
+        }
+    }))
+
+    return useObserver(() => (
+        <h1 onClick={todo.toggle}>
+            {todo.title} {todo.done ? "[DONE]" : "[TODO]"}
+        </h1>
+    ))
+})
+```
+
+When using `useLocalStore`, all properties of the returned object will be made observable automatically, getters will be turned into computed properties, and methods will be bound to the store and apply mobx transactions automatically.
+
+_Note: using `useLocalStore` is mostly beneficial for really complex local state, or to obtain more uniform code base. Note that using a local store might conflict with future React features like concurrent rendering._
```

Comment:
Does it conflict with concurrent mode? I mean it's `useState` underneath so it should be pretty safe. It might leak one observable I suppose, but that shouldn't be such an issue. I wouldn't probably warn about something we are not sure about yet.

Reply:
Afaik properly supporting concurrent mode means that it should be possible to rebase local state changes, which is not possible with this solution. Afaik it is not more than honest to warn that this is potientially not compatible with certain future react features, even when I'm not fully vested on their benefits in the general case yet :)

Reply:
Well, `useObserver` also wasn't Concurrent mode ready and we've figured it out. Such a warning feels premature and tells: "you better not to use this, otherwise, you will have to refactor later".

Reply:
I understand that nothing is certain, but as a user, I would much prefer see this in the readme too :)